google_guest_agent/agentcrypto/mtls_mds_windows.go

```diff
@@ -119,22 +176,30 @@ func (j *CredsJob) writeClientCredentials(creds []byte, outputFile string) error
 		return fmt.Errorf("failed to store pfx cert context: %w", err)
 	}
 
-	// Best effort to cleanup previous certificates and keep only the latest.
-	// When the agent restarts, it will not have the previous context in memory.
-	// Therefore, it will not remove any old certificates that are not required.
-	// If there are multiple similar certificates, clients should try to use the one
-	// with the longest expiry date, as that will be the latest version of it.
-	// TODO: See if we can have a more robust process for cleaning up old certificates,
-	// while ensuring that we don't delete any that were not installed by the agent process.
+	// Search for previous certificate if its not already in memory.
+	if prevCtx == nil && num != "" {
```

Comment:
Knowing if we could get the previously added cert in the certstore is a strong requirement - I'd say that cleanup + addition are atomic operations - I'd suggest moving the "find certificate" to the very beginning, with that if we fail to find it we can abort before we've written the new certificate, that way we can try again the whole process without dealing with inconsistent state.

Reply:
We talked offline, it seems to be fine.